android/PhysicalWeb/app/src/main/java/org/physical_web/physicalweb/UrlDeviceDiscoveryService.java

```diff
@@ -482,21 +483,39 @@ public long getScanStartTime() {
     return mScanStartTime;
   }
 
-  public void restartScan() {
-    for (UrlDeviceDiscoverer urlDeviceDiscoverer : mUrlDeviceDiscoverers) {
-      urlDeviceDiscoverer.stopScan();
+  public void startScan() {
+    if (mScanStartTime == 0) {
+      mScanStartTime = new Date().getTime();
     }
-    mScanStartTime = new Date().getTime();
     for (UrlDeviceDiscoverer urlDeviceDiscoverer : mUrlDeviceDiscoverers) {
       urlDeviceDiscoverer.startScan();
     }
   }
 
+  public void stopScan() {
+    for (UrlDeviceDiscoverer urlDeviceDiscoverer : mUrlDeviceDiscoverers) {
+      urlDeviceDiscoverer.stopScan();
+    }
+  }
+  public void restartScan() {
+    stopScan();
+    mScanStartTime = new Date().getTime();
+    startScan();
+  }
+
   public boolean hasResults() {
     return !mPwCollection.getGroupedPwPairsSortedByRank().isEmpty();
   }
 
   public PhysicalWebCollection getPwCollection() {
     return mPwCollection;
   }
+
+  public void clearCache() {
+    stopScan();
+    mScanStartTime = 0;
+    Utils.setPwsEndpoint(this, mPwCollection);
+    mPwCollection.clearCache();
+    saveCache();
```

Comment:
If that is done, when the user goes back to the nearby fragment, it may appear that the list didn't refresh at all. Is it better for the user to see that it is scanning again or just have results appear?